### PR TITLE
Charts

### PR DESCRIPTION
**Pull request by martin-ebay**
## Description
adding the 5 new chart components, area-chart, bar-chart, line-graph, trend-line and donut-chart (with donut-graph / chart legend sub components)

### CODE REVIEW

File: src/components/ebay-area-chart/component.js
Comment (agliga) [inline src/components/ebay-area-chart/component.js:29]: Do not assign something to state if we are not changing it
Use either `this` or set it as a getter or something

File: src/components/ebay-bar-chart/component.js
Comment (agliga) [inline src/components/ebay-bar-chart/component.js:236]: Use a token here

**Review by agliga — CHANGES_REQUESTED** [review]
So couple of comments
Some of these apply to all chart components (like readme changes and doc changes)
Need to add more comments what things do

But regardless I thought we had discussed to remove donut chart for now and I see it here. 
We can't merge it until skin is merged. We need to have it split up

Please address these. We plan on releasing soon so if we want to get this in we need these changes.

File: src/common/charts/donut-chart.js
Comment (ArtBlue) [reply src/common/charts/donut-chart.js:1]: That was my impression too. I thought we were skipping donut for now.

File: src/components/ebay-bar-chart/component.js
Comment (ArtBlue) [inline src/components/ebay-bar-chart/component.js:248]: This is a great example of what I was referring to in my previous comment about caching `this` to a meaningful const name. I would go bit further to explain what `refPoint` is, either make the const even more meaningful or add a comment.

File: src/components/ebay-bar-chart/component.js
Comment (ArtBlue) [inline src/components/ebay-bar-chart/component.js:70]: This shouldn't be static. We should get it from some central global config.

**Comment by martin-ebay** [comment]
let me see what happened cause this isn't matching my local
seems pushing to this branch from another didn't do as I expected
it's like it did a merge of them.

**Pull request closed without merging**
